tools/state-viewer/src/state_dump.rs

```diff
@@ -348,12 +373,94 @@ mod test {
+        let tx00 = SignedTransaction::from_actions(
+            1,
+            "test0".parse().unwrap(),
+            "test0".parse().unwrap(),
+            &signer0,
+            vec![Action::DeployContract(DeployContractAction {
+                code: near_test_contracts::base_rs_contract().to_vec(),
+            })],
+            genesis_hash,
+        );
+        env.clients[0].process_tx(tx00, false, false);
+        let tx01 = SignedTransaction::stake(
+            1,
+            "test0".parse().unwrap(),
+            &signer0,
+            TESTING_INIT_STAKE,
+            signer0.public_key.clone(),
+            genesis_hash,
+        );
+        env.clients[0].process_tx(tx01, false, false);
+
+        let signer1 =
+            InMemorySigner::from_seed("test1".parse().unwrap(), KeyType::ED25519, "test1");
+        let tx1 = SignedTransaction::stake(
+            1,
+            "test1".parse().unwrap(),
+            &signer1,
+            TESTING_INIT_STAKE,
+            signer1.public_key.clone(),
+            genesis_hash,
+        );
+        env.clients[0].process_tx(tx1, false, false);
+
+        safe_produce_blocks(&mut env, 1, epoch_length * 2 + 1);
+
+        let head = env.clients[0].chain.head().unwrap();
+        let last_block_hash = head.last_block_hash;
+        let cur_epoch_id = head.epoch_id;
+        let block_producers = env.clients[0]
+            .runtime_adapter
+            .get_epoch_block_producers_ordered(&cur_epoch_id, &last_block_hash)
+            .unwrap();
+        assert_eq!(
+            block_producers.into_iter().map(|(r, _)| r.take_account_id()).collect::<HashSet<_>>(),
+            HashSet::from_iter(vec!["test0".parse().unwrap(), "test1".parse().unwrap()])
+        );
+        let last_block = env.clients[0].chain.get_block(&head.last_block_hash).unwrap();
+        let state_roots: Vec<CryptoHash> =
+            last_block.chunks().iter().map(|chunk| chunk.prev_state_root()).collect();
+        let runtime = NightshadeRuntime::test(Path::new("."), store, &genesis);
+        let records_file = tempfile::NamedTempFile::new().unwrap();
+        let select_account_ids = vec!["test1".parse().unwrap()];
+        let new_near_config = state_dump(
+            runtime,
+            &state_roots,
+            last_block.header().clone(),
+            &near_config,
+            Some(&records_file.path().to_path_buf()),
+            Some(&select_account_ids),
+        );
+        let new_genesis = new_near_config.genesis;
+        let mut expected_accounts: HashSet<AccountId> =
+            new_genesis.config.validators.iter().map(|v| v.account_id.clone()).collect();
+        expected_accounts.extend(select_account_ids.clone());
+        for record in new_genesis.records.0.iter() {
+            if let StateRecord::Account { account_id, .. } = record {
+                assert!(expected_accounts.contains(account_id));
```

Comment:
I'd compare the new_genesis_records with expected_accounts.

(otherwise, if new_genesis has no accounts -- then your current code would still pass)

Reply:
Done!